jupyter-org-extensions.el

```diff
@@ -158,16 +165,17 @@ With prefix arg NEW, always insert new cell."
   (org-babel-execute-buffer))
 
 ;;;###autoload
-(defun jupyter-org-jump-to-block (&optional N)
+(defun jupyter-org-jump-to-block (&optional context)
   "Jump to a block in the buffer.
-If narrowing is in effect, only a block in the narrowed region.
-Use a numeric prefix N to specify how many lines of context to use.
+If narrowing is in effect, jump to a block in the narrowed region.
+Use a numeric prefix CONTEXT to specify how many lines of context to use.
 Defaults to 3."
   (interactive "p")
   (unless (require 'ivy nil t)
     (error "Package `ivy' not installed"))
   (let ((p '()))
-    (when (= 1 N) (setq N 3))
+    (when (= 1 context)
+      (setq context jupyter-org-jump-to-block-context-lines))
```

Comment:
Actually we should set this in the interactive spec with something like

```emacs-lisp
(interactive
 (list (if context (prefix-numeric-value context)
         jupyter-org-jump-to-block-context-lines)))
```

Then here we would have

```emacs-lisp
(or context (setq context jupyter-org-jump-to-block-context-lines))
```

The way we have it now would actually fail if `context` is `nil` since you can't compare `nil` with a number.

Reply:
I don't think this is quite needed:
The interactive declaration I'm using [i.e., `(interactive "p") ] already does the equivalent of `prefix-numeric-value`. In theory, `context` can only be a number.

Reply:
I am thinking more of the non-interactive case. You are right that the `interactive` spec doesn't need to change, but the function itself specifies that `context` is optional which means that it can be `nil` if called directly. For example calling `(jupyter-org-jump-to-block)` directly is valid and in this case, `context` will be `nil`. So really the issue is that a `nil` `context` needs to be handled. Changing the `interactive` spec really has nothing to do with the issue, sorry for the noise.

Also I messed up the interactive spec, it should be

```emacs-lisp
(interactive
 (list (if current-prefix-arg (prefix-numeric-value current-prefix-arg)
         jupyter-org-jump-to-block-context-lines)))
```

Reply:
How about having the following change instead?
```lisp
(when (or (= 1 context) (not context))
      (setq context jupyter-org-jump-to-block-context-lines))
```

Reply:
OK, I think this is where changing the interactive spec has benefits since it doesn't rely on needing to handle the special case of `context` being 1. With this change, if I where to evaluate 
`(jupyter-org-jump-to-block 1)` it would actually use `jupyter-org-jump-to-block-context-lines` for `context` instead of 1 line. If we change the interactive spec, then we don't need to handle this special case in the actual function.

In the function body, we don't know if the function has been called interactively or not, we could use `called-interactively-p` to check but changing the interactive spec is the more standard approach in this case.

Reply:
It's intentional that `(jupyter-org-jump-to-block 1)` sets the context to `jupyter-org-jump-to-block-context-lines` because there is no point in the context being `1`. If that were the case, only the `#+SRC_BLOCK` line would be shown without a proper context.
On the other hand, I'd in theory have to check if the context is below 1 and change it to `jupyter-org-jump-to-block-context-lines`. I'm therefore thinking that this change would also make sense:
```lisp
(when (or (< context 1) (not context))
      (setq context jupyter-org-jump-to-block-context-lines))
```
What do you think about this?

Reply:
> there is no point in the context being 1. If that were the case, only the #+SRC_BLOCK line would be shown without a proper context.

In that case, should the context mean the number of actual source lines?



If we keep with a `context` of 1 actually meaning `jupyter-org-jump-to-block-context-lines`, then if I do `(jupyter-org-jump-to-block 2)`, it will actually use a number of lines less than the default value of `jupyter-org-jump-to-block-context-lines` (assuming it is 3). Should this be possible?

I agree we should handle the case of the context being less than 1.

Reply:
I've integrated your suggestions into what I believe is the better solution: a custom interactive spec and a bounds check. The CONTEXT variable now means "lines of code after the header of the source block".